src/krux/pages/home_pages/mnemonic_xor.py

```diff
@@ -0,0 +1,164 @@
+
+        if not all(isinstance(s, str) for s in args):
+            raise TypeError("Inputs aren't str")
+
+        allowed_len = len(args[0].split(" "))
+        if not all(len(s.split(" ")) == allowed_len for s in args):
+            raise ValueError("Mnemonics should have same length")
+
+        entropies = []
+
+        # Check for low entropy before XOR
+        for s in args:
+            entropy = mnemonic_to_bytes(s)
+            allzero = len(entropy) * b"\x00"
+            allfull = len(entropy) * b"\xff"
+
+            # Is either of the mnemonic entropies all 0x00 bytes
+            # (then this will result: "mnemonic unchanged")
+            if entropy == allzero:
+                raise ValueError("Low entropy mnemonic")
+
+            # or is either of the mnenonics all 0xff bytes
+            # (then inverted entropy is result: "mnemonic inverted")
+            if entropy == allfull:
+                raise ValueError("Low entropy mnemonic")
+
+            entropies.append(entropy)
+
+        new_entropy = MnemonicXOR.xor_bytes(*entropies)
+        allzeros = len(entropies[0]) * b"\x00"
+        allfull = len(entropies[0]) * b"\xff"
+        if new_entropy in (allzeros, allfull):
+            raise ValueError("Low entropy mnemonic")
+
+        return mnemonic_from_bytes(new_entropy)
+
+    def export(self):
+        """Menu for XOR the current mnemonic with a share"""
+        menu = Menu(
+            self.ctx,
+            [
+                (t("Via Camera"), self.load_key_from_camera),
+                (t("Via Manual Input"), self.load_key_from_manual_input),
+                (t("From Storage"), self.load_mnemonic_from_storage),
+            ],
+        )
+
+        _, status = menu.run_loop()
+        return status
+
+    def _load_key_from_words(self, words, charset=LETTERS, new=False):
+        """
+        Similar method from krux.pages.login.Login without loading a key,
+        instead, it add the bytes from a mnemonic's entropy to the list of entropies
+        """
+        self.ctx.display.clear()
+        self.ctx.display.draw_centered_text(
+            t("The following mnemonic is the choosen share")
+            + ". "
+            + t("Review it before XOR with the current mnemonic")
+            + "."
+        )
+
+        if not self.prompt(t("Proceed?"), BOTTOM_PROMPT_LINE):
+            return MENU_CONTINUE
+
+        from ..mnemonic_editor import MnemonicEditor
+        from ...key import Key
+        from ...wallet import Wallet
+
+        # Show share before XOR
+        mnemonic = MnemonicEditor(self.ctx, " ".join(words)).edit()
+
+        # XOR mnemonics
+        xored_mnemonic = MnemonicXOR.xor_mnemonics(
+            self.ctx.wallet.key.mnemonic, mnemonic
+        )
+
+        # Show derived mnemonic
+        self.ctx.display.clear()
+        self.ctx.display.draw_centered_text(
+            t("The following mnemonic is the XORed mnemonic")
```

Comment:
Suggestion:
"Resulting XORed mnemonic. Loading it will replace original mnemonic and discard any passphrase or loaded descriptor"